Update looks and animations of character cards in gallery

ye7iaserag opened the pull request:
This brings an appealing 3d look to gallery extension character cards.
It:
- Adds hover animations.
- Adds highlighting to selected character.
- Handles initialization and pagination for a preselected character.

How it should look:
![Untitled](https://user-images.githubusercontent.com/2270440/229264268-b20d1d70-2902-4fdb-abfb-163404e04e8a.png)

ye7iaserag commented:
@oobabooga do you think this can be merged? it's low risk and shouldn't cause problems while making chat mode look more like a dedicated tool for chat

oobabooga commented:
@ye7iaserag I like the slightly increased character card size for selected characters, but I found the rotation of the character cards unnecessary. Also, the shadow seems to extend a bit too far/become too fuzzy.

Do you think that you can make a compromise between the current gallery and your PR version?

ye7iaserag commented:
I raise you a compromise.

Changes:
- Lowered the excessive shadow on selection
- Added 3d effect to card on hover based on mouse position
- Added glow effect on hover

Now it screams "card gallery", makes you wanna... "catch em all" xD

If you think it's excessive, I can dail it down or even close the PR all together

mcmonkey4eva commented on extensions/gallery/script.py:
```diff
@@ -49,29 +51,132 @@ def generate_css():
+
+      init();
+      update();
+      //alert("JS Inserted");
```
probably should remove this debug.

mcmonkey4eva commented:
Somewhat external to the PR since it wasn't this PR that added the original code for it, but:
Probably the `css` and `js` content should be in their own files, and linked via file ref (I think `/file/extensions/gallery/style.css` is the proper path?)
Cleaner/nicer to deal with that in the middle of a python file.

ye7iaserag commented:
@mcmonkey4eva I really wanted to do this but gradio does not support serving static files.
The only solution that comes to mind has to come from the integration APIs here, and that would be to:
- Allow each extension to register static files on start up.
- Inject those files in document head.
- Host a dedicated FastApi server to serve those static files.
- Create a helper function to be able to get absolute URLs to those resources.

Which is not the goal of this PR.

mcmonkey4eva commented:
Gradio does serve static files, via sub-URLs like the one I gave in my message there.

ye7iaserag commented:
Ok thanks a lot, will try that later since I'm working on https://github.com/oobabooga/text-generation-webui/discussions/973

If you ran the code, I'd like to hear your opinion on the looks

ye7iaserag commented:
Moved the files outside of script to a dedicated resources folder
Only left initial JS because if it's added to HTML DOM directly it will not work (HTML5 feature), so it has to be injected through JS

@oobabooga please check this

oobabooga commented:
@ye7iaserag I feel like this adds a lot of complexity to the extension while not making the looks necessarily better. I'd prefer to keep the current implementation (which was made possible by you!).

ye7iaserag commented:
@oobabooga  Ok no problem.
Is there a way to host a listing of external extensions in this repo? (Just the urls of other repos to give visibility)
SD webui hosts a listing in a separate markdown file

oobabooga commented:
I think that you refer to https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/blob/master/index.json

I don't have that at the moment, but if you want to create your own gallery extension (or any other), you can give me the link and I'll put it in https://github.com/oobabooga/text-generation-webui/wiki/Extensions
